### ISSUE TITLE
Add sessionAttributes to "Session Start" event

### ISSUE THREAD

**Issue by meikitanious** [mParticle/mparticle-web-sdk#830]
## Problem:

An eCommerce client of mine would like to start associating traits with a session, such as utm parameters. These session attributes would be available before the SDK initializes. However, the `session_start` event does not pass session attributes.

The `session_start` event does have a `custom_attributes` key in its payload, however it's left as a null value.

This is an issue because while we do receive an event for every new session, we do not receive an event for every time a session is terminated. We only receive session end events when a session times out, and not when a user exits the browser.

Here is an example of their session start events from the last 7 days.

![Screenshot 2024-01-09 at 4 31 05 PM](https://github.com/mParticle/mparticle-web-sdk/assets/98788089/4ee8d1db-9563-42ed-b315-b9375d53ec15)

While this is an example of their session end events. 

![Screenshot 2024-01-09 at 4 31 48 PM](https://github.com/mParticle/mparticle-web-sdk/assets/98788089/d3cbaf25-e84d-4256-bfe3-00b8d0cda6ea)

### Using a Custom Event (Current Solution)

I have a custom event called "Session Data" which is an event that manually passes these parameters. However, this is harder to query in the warehouse and is an unnecessary amount of extra events.

### Editing sessionStorage values before the event is batched

I tried writing a script that listens to changes in localStorage based on the key where the eventbatch data is stored. This did not work either.

### Adding Global Event Attributes / Flags in Config (Ideal)

Having the ability to add customAttributes and customFlags  in our config that get applied to **all events** would make the most sense.  Here is why this option is preferable:

1. It reduces redundancy in manually applying customFlags / customAttributes that are shared across multiple event types.
2. Because it's in our configuration, it means those attributes should be applied to the session_start event without needing a delay to read sessionAttributes from storage.

### A config option for sessionAttributes

Basically a section in the config where we can pass an object, similar to customFlags, that will add sessionAttributes to the session_start event.

**Comment by alexs-mparticle** [comment]
@meikitanious Thanks for providing such a detailed use case. It's amazingly helpful. 

You are right about sessions ending on timeouts, and not when a user exits the browser. Our session docs - https://docs.mparticle.com/developers/sdk/web/session-management/#session-timeout detail one other case.  If a user returns to a browser beyond the session timeout, we will send the session end event at that point.

You may also be able to listen for a [beforeunload event](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event) and force a Session End, but this is not completely reliable on mobile devices.

I recommend having your client reach out to their Customer Success Manager or visiting our [Support Page to open a ticket](https://support.mparticle.com/) so that your feature request can get prioritized.

**Comment by meikitanious** [comment]
Thank you for the quick reply @alexs-mparticle. I reached out to Tiffany L. regarding this yesterday.

I'd also be happy to open a PR for this if I could get pointed in the right direction. Do you know what file or directory in the web SDK deals with the NewSession function on initialization? My understanding is it happens after the logic check of seeing if there's an existing session.

**Comment by alexs-mparticle** [comment]
Closing this out due to inactivity. If this is still an issue, please feel free to reopen or create a new issue.
